### DISCUSSION TITLE
search_url() templates/oscar/results.html  genarate url ignoredsettings['server']['base_url']

### DISCUSSION THREAD

**Discussion by jdxin0** [searx/searx#2547] [Installation issue: custom]
**Version of Searx, commit number if you are using on master branch and stipulate if you forked Searx**

`0.18.0-160-ab873980`

**How did you install Searx?**
nignx
```
    server {
        server_name  domian.org;

	location / {
	    proxy_pass http://127.0.0.1:8080/;
       }
    }
```

docker
```base
docker run -it -p 8080:8080 -e BASE_URL=https://domian.org/ searx/searx
```

**What happened?**
Search Url is "127.0.0.1:8080"
<img width="296" alt="image" src="https://user-images.githubusercontent.com/11607613/107196498-a5a41a80-6a2d-11eb-9f72-421afd11d463.png">

**How To Reproduce**
1. configure nignx as above
2. start docker
3. search anything

**Expected behavior**
Search Url is "domain.org"
<img width="296" alt="image" src="https://user-images.githubusercontent.com/11607613/107196556-bce30800-6a2d-11eb-9730-5d2d95ff2c53.png">

**Screenshots & Logs**
<img width="296" alt="image" src="https://user-images.githubusercontent.com/11607613/107196498-a5a41a80-6a2d-11eb-9f72-421afd11d463.png">

**Additional context**

in [templates/oscar/results.html ](https://github.com/searx/searx/blob/709dd960f125af17cffa10715d5bdef15018a918/searx/templates/oscar/results.html) line 10

```python
{%- macro search_url() %}{{ url_for('search', _external=True) }}?q={{ q|urlencode }}{% if selected_categories %}&amp;categories={{ selected_categories|join(",") | replace(' ','+') }}{% endif %}{% if pageno > 1 %}&amp;pageno={{ pageno }}{% endif %}{% if time_range %}&amp;time_range={{ time_range }}{% endif %}{% if current_language != 'all' %}&amp;language={{ current_language }}{% endif %}{% endmacro -%}
```

with `_external=True`, flask `url_for` will generate full url, the action to urljoin `settings['server']['base_url'] `will be invalid.

```python
def url_for_theme(endpoint, override_theme=None, **values):
    if endpoint == 'static' and values.get('filename'):
        theme_name = get_current_theme_name(override=override_theme)
        filename_with_theme = "themes/{}/{}".format(theme_name, values['filename'])
        if filename_with_theme in static_files:
            values['filename'] = filename_with_theme
    url = url_for(endpoint, **values)
    if settings['server']['base_url']:
        if url.startswith('/'):
            url = url[1:]
        url = urljoin(settings['server']['base_url'], url)
    return url
```

```python
>>> settings['server']['base_url']
'https://xxx.xxx.org/'
>>> urljoin(settings['server']['base_url'], 'http://127.0.0.1:8080/search')
'http://127.0.0.1:8080/search'
```

**Comment by unixfox** [comment]
You didn't pass the correct headers on NGINX.

Change your config to:
```
server {
        server_name  domian.org;

	location / {
	    proxy_pass http://127.0.0.1:8080/;
      proxy_set_header   Host             $http_host;
    proxy_set_header   Connection       $http_connection;
    proxy_set_header   X-Real-IP        $remote_addr;
    proxy_set_header   X-Forwarded-For  $proxy_add_x_forwarded_for;
    proxy_set_header   X-Scheme         $scheme;
       }
    }
```

And restart nginx. You can find this NGINX example configuration on our wiki: https://searx.github.io/searx/admin/installation-nginx.html#the-nginx-http-server

**Reply by jdxin0** [reply]
Got it. 
Both methods are ok.
1. templates/oscar/results.html line 10, remove `_external=True`
2. change nginx settings

**Reply by unixfox** [reply]
Well not really, the second method is the one to chose because the first one will break any further update of searx if something in the code got modified in the file `templates/oscar/results.html`.
